mempool/src/mempool.rs

```diff
@@ -289,6 +289,23 @@ impl Mempool {
             .abort();
     }
 
+    /// This function prunes transactions from the mempool that were already included in the blockchain
+    /// This is generally used by history sync when it adopts epochs
+    pub fn mempool_clean_up(&self, _: &Blake2bHash) {
```

Comment:
The argument is unused. This opens up the question on why do we need it to be sent with the event?

Reply:
Is used by the _get_newest_hash_ function also to be consistent with the other events in case it is needed in the future

Reply:
I'd propose then to remove the unused argument in the mempool and then change
`on_blockchain_history_adopted(&mut self, hash: &Blake2bHash)` for `on_blockchain_history_adopted(&mut self, _: &Blake2bHash)`